libp2p/pubsub/gossipsub.py

```diff
@@ -324,15 +417,15 @@ def _get_peers_to_send(
                     topic, self.degree - num_mesh_peers_in_topic, self.mesh[topic]
                 )
 
-                fanout_peers_not_in_mesh: List[ID] = [
-                    peer for peer in selected_peers if peer not in self.mesh[topic]
-                ]
-                for peer in fanout_peers_not_in_mesh:
+                for peer in selected_peers:
                     # Add peer to mesh[topic]
                     self.mesh[topic].append(peer)
 
                     # Emit GRAFT(topic) control message to peer
-                    await self.emit_graft(topic, peer)
+                    if peer not in peers_to_graft:
+                        peers_to_graft[peer] = [topic]
```

Comment:
Would it be better to use `defaultdict` for `peers_to_graft`?

Reply:
Good idea